presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDriverUri.java

```diff
@@ -83,9 +88,14 @@ public URI getHttpUri()
         return buildHttpUri();
     }
 
+    public Properties getConnectionProperties()
+    {
+        return connectionProperties;
+    }
+
     private static Map<String, String> parseParameters(String query)
     {
-        Map<String, String> result = new HashMap<>();
+        Map<String, String> result = new HashMap<>(ConnectionProperties.getDefaults());
```

Comment:
Why do you need `defaults` here? This method is supposed to parse only the query string. You can merge it with the defaults later.